Clarify Server/Message Management Description (#4437)
hammerandchisel/discord-api-docs: docs/topics/Permissions.md

| BAN_MEMBERS \*                | `0x0000000000000004` `(1 << 2)`  | Allows banning members                                                                                                                              |              |
| ADMINISTRATOR \*              | `0x0000000000000008` `(1 << 3)`  | Allows all permissions and bypasses channel permission overwrites                                                                                   |              |
| MANAGE_CHANNELS \*            | `0x0000000000000010` `(1 << 4)`  | Allows management and editing of channels                                                                                                           | T, V, S      |

> | MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management of the server name, region, and adding bots to the server                                                                          |              |

Reviewer: server region doesn't exist anymore, and manage server does a lot more than what is listed here.

Author: Fair. I'm updating that in my latest commit involving most of the suggestions made here

Author: I would rather err on the side of being more specific then more vague, let me know if there is anything still missing from manage server in the latest commit.

> | MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management of the server name, region, and adding bots to the server                                                                          |              |

Reviewer 2: ```suggestion
| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows deleting invites, editing guild settings and welcome screen and managing guild template, integrations and widget                             |              |
```

Reviewer 3: "editing guild settings" *includes* welc screen, template, widget and integrations, i think its just easier to leave it as Allows management and editing of the guild

Reviewer 2: Editing [guild settings](https://discord.com/developers/docs/resources/guild#modify-guild) and [welcome screen](https://discord.com/developers/docs/resources/guild#modify-guild-welcome-screen) and managing [guild template](https://discord.com/developers/docs/resources/guild-template#modify-guild-template), [integrations](https://discord.com/developers/docs/resources/guild#delete-guild-integration) and [widget](https://discord.com/developers/docs/resources/guild#modify-guild-widget) are all different things in the term of API, they use clearly *different* endpoints, and my suggestion reflects them, assuming OP's intention is to clarify things.

Author: > Editing [guild settings](https://discord.com/developers/docs/resources/guild#modify-guild) and [welcome screen](https://discord.com/developers/docs/resources/guild#modify-guild-welcome-screen) and managing [guild template](https://discord.com/developers/docs/resources/guild-template#modify-guild-template), [integrations](https://discord.com/developers/docs/resources/guild#delete-guild-integration) and [widget](https://discord.com/developers/docs/resources/guild#modify-guild-widget) are all different things in the term of API, they use clearly _different_ endpoints, and my suggestion reflects them, assuming OP's intention is to clarify things.

Is there a way to insert those links into the docs so people can see those descriptions of what each of those things entails?

| PRIORITY_SPEAKER              | `0x0000000000000100` `(1 << 8)`  | Allows for using priority speaker in a voice channel                                                                                                | V            |
| STREAM                        | `0x0000000000000200` `(1 << 9)`  | Allows the user to go live                                                                                                                          | V            |
| VIEW_CHANNEL                  | `0x0000000000000400` `(1 << 10)` | Allows guild members to view a channel, which includes reading messages in text channels                                                            | T, V, S      |
| SEND_MESSAGES                 | `0x0000000000000800` `(1 << 11)` | Allows for sending messages in a channel (does not allow sending messages in threads)                                                               | T            |
| SEND_TTS_MESSAGES             | `0x0000000000001000` `(1 << 12)` | Allows for sending of `/tts` messages                                                                                                               | T            |

> | MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of or pinning other users messages                                                                                             | T            |

Reviewer: ```suggestion
| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of other users' messages and pinning messages                                                                                   | T            |
```

Reviewer 4: it also allows bypassing slowmode and publishing all messages in news channels

Reviewer 5: and removing reactions

Reviewer: yeah... maybe this should just say "managing messages"

Author: > yeah... maybe this should just say "managing messages"

I would rather be more specific then that, since someone may be looking in these documents for specific permissions to request for a specific task (such as deleting reactions) and I would like that to be supported as fully as is reasonably possible in the documentation.

> | MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of or pinning other users messages                                                                                             | T            |

Reviewer 2: ```suggestion
| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows crossposting other users' messages in news channels, deleting other users' messages and reactions and pinning messages                       | T            |
```

Author: I'm manually adding this since it conflicted with another suggestion.

| BAN_MEMBERS \*                | `0x0000000000000004` `(1 << 2)`  | Allows banning members                                                                                                                              |              |
| ADMINISTRATOR \*              | `0x0000000000000008` `(1 << 3)`  | Allows all permissions and bypasses channel permission overwrites                                                                                   |              |
| MANAGE_CHANNELS \*            | `0x0000000000000010` `(1 << 4)`  | Allows management and editing of channels                                                                                                           | T, V, S      |

> | MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows deleting invites, editing guild settings, guild welcome screen, and managing a guild's template, integrations (bots, webhooks, and app integrations), and the guild widget                             |              |

Reviewer 3: ```suggestion
| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management and editing of the guild                                                               |              |
```

Author: @Jupith, since editing an management of the guild isn't terribly helpful for someone who doesn't already know what that means, I feel that something more descriptive would be more beneficial. Unless there is an issue with formatting, is there a good reason I'm missing for shortening the description?

Reviewer 4: There could be a section describing what each extended permission does (by that i mean the ones that give access to multiple things, like Manage Server/Messages) so the table doesn't become too big, similar to [this](https://discord.com/developers/docs/topics/permissions#implicit-permissions)

Author: @Suspense4615  What if it were changed to `Allows managing invites, settings, welcome screen, template, integrations (bots, webhooks, and app integrations), and the widget of a guild`? Then its about the same length as the description in the moderate members permission. It seems like getting up to three lines isn't a huge deal, and the added clarity would be a big improvement, without having to create a new section.

Author: Alternatively, if we used @PlavorSeol 's suggestion from above, (quoted below) and had each of his categories link to the relevant endpoint information, then that would shorten things and ensure detailed descriptions

He suggested we change it to `Allows deleting invites, editing guild settings and welcome screen and managing guild template, integrations and widget`

His comment on that was:
> Editing [guild settings](https://discord.com/developers/docs/resources/guild#modify-guild) and [welcome screen](https://discord.com/developers/docs/resources/guild#modify-guild-welcome-screen) and managing [guild template](https://discord.com/developers/docs/resources/guild-template#modify-guild-template), [integrations](https://discord.com/developers/docs/resources/guild#delete-guild-integration) and [widget](https://discord.com/developers/docs/resources/guild#modify-guild-widget) are all different things in the term of API, they use clearly _different_ endpoints, and my suggestion reflects them, assuming OP's intention is to clarify things.

Reviewer 4: This is not taking upcoming features into account, while it fits with 3 lines *now*, Discord might add several features tied to that permission in the future. And I don't think adding a few lines once every now and then is sustainable on the long run

Author: @Suspense4615 Can you show me an example of what you think would work best?

Reviewer 4: > @Suspense4615 Can you show me an example of what you think would work best?

something like [this](https://discord.com/developers/docs/topics/permissions#implicit-permissions)
